quic/s2n-quic-core/src/buffer/receive_buffer/slot.rs

```diff
@@ -128,6 +136,44 @@ impl Slot {
     pub fn end_allocated(&self) -> u64 {
         self.end
     }
+
+    #[inline]
+    pub fn skip(&mut self, len: u64) {
+        // trim off the data buffer
+        unsafe {
+            let len = len.min(usize::MAX as u64) as usize;
```

Comment:
why take a `u64` for `len` instead of `usize`?

Reply:
The caller is using a `u64` and would have to convert to `usize` anyway. The len should never be bigger that 64k so it probably doesn't matter either way. But I can change it to a usize if you think that's clearer.

Reply:
I saw the `add_start` above using `usize` so I was currious, but as you said it doesn't really matter. If it should never be bigger than 64k should we put a debug_assert! or do something like `let len:usize = len.try_into().expect("len must be <= 64")`

Reply:
I'll try adding an assertion and see if my assumption is true 😁